media-source/mediasource-activesourcebuffers.html

```diff
@@ -0,0 +1,212 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <title>Checks MediaSource.activeSourceBuffers and changes to selected/enabled track state</title>
+        <script src="/resources/testharness.js"></script>
+        <script src="/resources/testharnessreport.js"></script>
+        <script src="mediasource-util.js"></script>
+    </head>
+    <body>
+        <div id="log"></div>
+        <script>
+          // Audio / Video files supported by the user agent under test
+          var subType = MediaSourceUtil.getSubType(MediaSourceUtil.AUDIO_ONLY_TYPE);
+          var manifestFilenameAudio = subType + "/test-a-128k-44100Hz-1ch-manifest.json";
+          var manifestFilenameVideo = subType + "/test-v-128k-320x240-30fps-10kfr-manifest.json";
+          var manifestFilenameAV = subType + "/test-av-384k-44100Hz-1ch-320x240-30fps-10kfr-manifest.json";
+
+          mediasource_test(function(test, mediaElement, mediaSource)
+          {
+              mediaElement.addEventListener("error", test.unreached_func("Unexpected event 'error'"));
+              MediaSourceUtil.fetchManifestAndData(test, manifestFilenameAudio, function (typeAudio, dataAudio)
+              {
+                  var sourceBuffer = mediaSource.addSourceBuffer(typeAudio);
+                  assert_equals(mediaSource.activeSourceBuffers.length, 0,
+                    "activeSourceBuffers is empty to start with");
+
+                  test.expectEvent(mediaElement, "loadedmetadata");
```

Comment:
It shouldn't matter much, but the actual sequence of these events being enqueued is: addsourcebuffer first, then loadedmetadata later once the first init segments across all sourcebuffers has appended and parsed. I'll swap the ordering as part of merging this.